### ISSUE TITLE
Dimensions.get('window').width defaults to screen width

### ISSUE THREAD

**Issue by jbcullis** [microsoft/react-native-macos#915]
### Environment

```markdown
react-native 0.64.0:
npm ls react-native-macos: 0.64.13
node: 17.0.1
npm: 8.1.0
yarn: N/A
xcodebuild: 13A1030d
```

### Steps to reproduce the bug

Load the app, before manually resizing the window call Dimensions.get('window').width and the return is the screen width, not app width.

After a manually resizing the window the return value is correct.

** FURTHER INFORMATION **
On initial load, the screen size is correct, and if you reload the app by using right click, reload it is also fine.
But if you reload the app by pressing r on the console then it grabs the screen dimensions rather than the window dimensions.

### Expected Behavior

Return dimension of app window.

### Actual Behavior

Return dimension of screen.

**Comment by HeyImChris** [comment]
Could you please post some sample code that we can use to easily repro this?

**Comment by jbcullis** [comment]
@HeyImChris On initial load, the screen size is correct, and if you reload the app by using right click, reload it is also fine.
But if you reload the app by pressing r on the console then it grabs the screen dimensions rather than the window dimensions.

I've been testing today and this is not an issue with the app in production, only presents when you use 'r' to reload the app. Feel free to close since there is an easy workaround using the right click - reload option but worth noting cause it tripped me up for a second.

**Comment by shirakaba** [comment]
That issue hit us today.

The original poster is talking about "screen width" and "app width" and I think is referring to:

- "screen width": the width of the main UIWindow's frame (which below I show as `windowRect`).
- "app width": the width of the content rect of the main UIWindow's frame (which below I show as `contentRect`).

```objc
// The rect of the UIWindow. This includes the title bar.
NSRect windowRect = self.window.frame;

// The content area within the UIWindow. This excludes the title bar.
NSRect contentRect = [self.window contentRectForFrameRect: windowRect];
```

Here it is illustrated:

![Image](https://github.com/user-attachments/assets/55f53cae-248c-4166-92c1-7a74ad40ba68)

So in other words it sounds like, `Dimensions.get("window").width` is initialised to `windowRect.frame.size.width` when instead it should be initialised to `contentRect.frame.size.width`... assuming the app fills the whole window, which unfortunately is not the case for brownfield apps.

I suppose the proper solution is to not take an initial value at all, and force a re-render to get the content rect's width?

**Comment by Saadnajmi** [comment]
Just for reference, here is where the (heavily ifdefed for macOS and visionOS...) code is that useWindowDimensions eventually calls:
https://github.com/microsoft/react-native-macos/blob/296b251e2486480d6db63f2addf9a25163438794/packages/react-native/React/UIUtils/RCTUIUtils.m#L12

**Comment by jbcullis** [comment]
For us, we just get zero values using the Dimension listener on macOS. All other platforms work ok.

We have a native module to cover the need:

#import "RNWindow.h"
#import <React/RCTEventEmitter.h>

@interface RNWindow : RCTEventEmitter <RCTBridgeModule>
@end

@implementation RNWindow
- (dispatch_queue_t)methodQueue{
  return dispatch_get_main_queue();
}


RCT_EXPORT_MODULE()

- (NSArray<NSString *> *)supportedEvents {
    return @[@"windowSizeChanged"];
}


- (void)startObserving {
    NSWindow *mainWindow = [[NSApplication sharedApplication] mainWindow];
    if (mainWindow) {
        [[NSNotificationCenter defaultCenter] addObserver:self
                                                 selector:@selector(handleWindowDidResize:)
                                                     name:NSWindowDidResizeNotification
                                                   object:mainWindow];
    }
}

- (void)stopObserving {
    [[NSNotificationCenter defaultCenter] removeObserver:self name:NSWindowDidResizeNotification object:nil];
}

- (void)handleWindowDidResize:(NSNotification *)notification {
    NSRect windowRect = [[notification object] frame];
    [self sendEventWithName:@"windowSizeChanged" body:@{@"width": @(windowRect.size.width), @"height": @(windowRect.size.height)}];
    //[self sendEventWithName:@"windowSizeChanged" body:@{@"width": @(windowRect.size.width)}];
  
}

RCT_EXPORT_METHOD(trackWindowSize) {
    [self startObserving];
}

// Function to get window width
RCT_EXPORT_METHOD(getWindowWidth:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
  dispatch_async(dispatch_get_main_queue(), ^{
    NSRect windowRect = [[NSApplication sharedApplication].mainWindow frame];
    resolve(@(windowRect.size.width));
  });
}

// Function to get window height
RCT_EXPORT_METHOD(getWindowHeight:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
  dispatch_async(dispatch_get_main_queue(), ^{
    NSRect windowRect = [[NSApplication sharedApplication].mainWindow frame];
    resolve(@(windowRect.size.height));
  });
}

RCT_EXPORT_METHOD(Resize:(NSInteger)width height:(NSInteger)height ) {
  
  NSRect frame;
  frame.size.height = (CGFloat)height;
  frame.size.width = (CGFloat)width;
  dispatch_async(dispatch_get_main_queue(),
                 ^{
    
    [[[[NSApplication sharedApplication] windows] objectAtIndex:0] setFrame:frame display:YES animate:NO];
    [[[[NSApplication sharedApplication] windows] objectAtIndex:0] center ];
    
  });
}

@end
